src/fmnoise/coldbrew.clj

```diff
@@ -181,11 +192,10 @@
                               :else 3))
         fdefn (filter some? (list name docstring args prepost))]
     `(do
-      (def ^:private ~fname
-         (cached
-          (with-meta
-            (fn [~@cache-key]
-              ~@body)
-            ~cache-options)))
-      (defn ~@fdefn (~fname ~@cache-key))
-      (vary-meta ~name merge ~(meta name)))))
+       (def ^:private ~fname (cached
+                              (with-meta
+                                (fn [~@cache-key]
+                                  ~@body)
+                                ~cache-options)))
+       (defn ~@fdefn (~fname ~@cache-key))
+       (alter-meta! (var ~name) merge (meta ~fname) ~(meta name)))))
```

Comment:
interesting, does `alter-meta!` fit better than `vary-meta` here?

Reply:
yes, vary-meta returns new object, same as with-meta, while alter-meta! does meta mutation, but it works on var reference level rather than the object level

Reply:
I don't remember whether I tested all possibilities, but as I remember, in this case, I could have used `(alter-meta! (var ~name) merge (meta ~fname))` because even without `name`, meta persisted well.